src/main/kotlin/com/exactpro/th2/template/BoxMain.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:JvmName("BoxMain")
+package com.exactpro.th2.template
+
+import com.exactpro.th2.common.event.Event
+import com.exactpro.th2.common.event.EventUtils
+import com.exactpro.th2.common.grpc.EventBatch
+import com.exactpro.th2.common.metrics.liveness
+import com.exactpro.th2.common.metrics.readiness
+import com.exactpro.th2.common.schema.factory.CommonFactory
+import mu.KotlinLogging
+import java.util.Deque
+import java.util.concurrent.ConcurrentLinkedDeque
+import java.util.concurrent.locks.Condition
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.concurrent.thread
+import kotlin.system.exitProcess
+
+private val LOGGER = KotlinLogging.logger { }
+
+fun main(args: Array<String>) {
```

Comment:
How about creating a class `Application` and using it here instead of gathering all initialization logic in main method?

```kotlin
class Application(
  private val commonFactory: CommonFactory,
  private val closeOnShutdown: (resourceName: String, action: () -> Unit) -> Unit,
) {
  // init logic
  fun run() {
    // run application
  }
}
```